hilti/runtime/src/types/bytes.cc

```diff
@@ -113,6 +113,12 @@ std::string Bytes::decode(bytes::Charset cs, bytes::DecodeErrorStrategy errors)
                         case DecodeErrorStrategy::IGNORE: break;
                         case DecodeErrorStrategy::REPLACE: s += "?"; break;
                         case DecodeErrorStrategy::STRICT: throw RuntimeError("illegal ASCII character in string");
+                        case DecodeErrorStrategy::ESCAPE: {
+                            std::stringstream stream;
+                            stream << std::setfill('0') << std::setw(2) << std::hex << (c|0);
+                            s += "\\x";
+                            s += stream.str();
```

Comment:
This duplicates what we already implement in `hilti::rt::escapeBytes`. Since it builds a new `stringstream` for each byte iterated over it might also become fairly expensive.

Let's instead use that library function before we create `s` above

```cpp
            if ( errors.value() == DecodeErrorStrategy::ESCAPE )
                return escapeBytes(str());
```
and mark this branch here as unreachable
```suggestion
                            cannot_be_reached();
```

Reply:
👍🏻